subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformCachingIntegrationTest.groovy

```diff
@@ -814,6 +796,249 @@ class ArtifactTransformCachingIntegrationTest extends AbstractHttpDependencyReso
+    }
+
+    @Unroll
+    def "failure in transformation chain propagates (position in chain: #failingTransform"() {
+        given:
+
+        Closure<String> possiblyFailingTransform = { index ->
+            index == failingTransform ? "FailingDuplicator" : "Duplicator"
+        }
+        buildFile << declareAttributes() << withJarTasks() << withLibJarDependency("lib3.jar") << duplicatorTransform << """
+            class FailingDuplicator extends Duplicator {
+                
+                @javax.inject.Inject
+                FailingDuplicator(int numberOfOutputFiles, boolean differentOutputFileNames) {
+                    super(numberOfOutputFiles, differentOutputFileNames) 
+                }                                                       
+
+                @Override
+                List<File> transform(File input) {
+                    throw new RuntimeException("broken")
+                }                
+            }
+
+            project(':app') {
+                dependencies {
+                    compile project(':lib')
+                }
+            }
+
+            allprojects {
+                dependencies {
+                    registerTransform {
+                        from.attribute(artifactType, "jar")
+                        to.attribute(artifactType, "green")
+                        artifactTransform(${possiblyFailingTransform(1)}) { params(2, true) }
+                    }
+                    registerTransform {
+                        from.attribute(artifactType, "green")
+                        to.attribute(artifactType, "blue")
+                        artifactTransform(${possiblyFailingTransform(2)})  { params(1, false) }
+                    }
+                    registerTransform {
+                        from.attribute(artifactType, "blue")
+                        to.attribute(artifactType, "yellow")
+                        artifactTransform(${possiblyFailingTransform(3)})  { params(3, false) }
+                    }
+                    registerTransform {
+                        from.attribute(artifactType, "yellow")
+                        to.attribute(artifactType, "orange")
+                        artifactTransform(${possiblyFailingTransform(4)})  { params(1, true) }
+                    }
+                }
+                task resolve {
+                    def artifacts = configurations.compile.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'orange') }
+                    }.artifacts
+                    inputs.files artifacts.artifactFiles
+
+                    doLast {
+                        println "files: " + artifacts.artifactFiles.collect { it.name }
+                        println "ids: " + artifacts.collect { it.id.displayName }
+                        println "components: " + artifacts.collect { it.id.componentIdentifier }
+                    }
```

Comment:
Is it useful to add these printouts everywhere? Maybe we could define a `ResolveTask` type that would have a property to assign the artifact view to. That would simplify these tests quite a bit.

Reply:
I added a `Resolve` task type and use it.